src/app/components/upload-files-modal/upload-files-modal.component.html

```diff
@@ -0,0 +1,66 @@
+
+<!-- 
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+-->
+<zlux-tab-trap></zlux-tab-trap>
+<!-- FA Icon is determined by class name, so we hardcode the style here -->
+<mat-icon style="font-size: 30px; position: absolute;">cloud_upload</mat-icon>
+<!-- Intentional lazy recycling of "create" modal CSS, TODO: move to shared/modal css -->
+<h2 mat-dialog-title class="modal-content-body">Upload Files in {{folderPath}}</h2>
+
+  <button mat-button class="modal-mat-button add" (click)="addFile()">Select File</button>
+
+  <div style="max-height: 400px; overflow-y:scroll;">
+    <ul>
+        <li class="" *ngFor="let file of files; index as i">
+            <label>
+                {{ file.name }}
+            </label>
+        
+            <mat-form-field appearance="standard" style="margin-left: 36px;">
+              <input matInput required type="text"
+                [(ngModel)]="selectedOption"
+                [matAutocomplete]="auto"
+                [errorStateMatcher]="matcher"
+                #encodingInput="ngModel"
+                placeholder="Select target encoding"
+              >
+              <mat-autocomplete autoActiveFirstOption #auto="matAutocomplete" [displayWith]="displayFn">
+                <mat-option *ngFor="let option of encodings" [value]="option.value">
+                  {{ option.name }}
+                </mat-option>
+              </mat-autocomplete>
+              <mat-error *ngIf="encodingInput.errors?.required">
+                Encoding is required
+              </mat-error>
+            </mat-form-field>
+            <a href="https://www.ibm.com/docs/en/zos/2.1.0?topic=ccsids-encoding-scheme" target="_blank">
+              <i class="fa fa-question-circle dataset-properties-question-circle"></i>
+            </a>
+        </li>
+    </ul>
+</div>
+
+<mat-dialog-actions>
+  <button mat-button mat-dialog-close class="modal-mat-button cancel">Cancel</button>
+  <button mat-button mat-dialog-close class="modal-mat-button create" (click)="uploadHandlerSetup()">Upload</button>
```

Comment:
Currently, I can click the Upload even if I haven't selected any file or not selected an encoding. Please add `[disabled]` attribute with an appropriate condition in order to enable the button only when the dialog is filled in correctly.

Reply:
it's fixed.